### DISCUSSION TITLE
Nuxt nitro node build served from express doesn't seem to work

### DISCUSSION THREAD

**Discussion by jeremeevans** [nuxt/nuxt#17897] [Questions]
I am trying to serve my Nuxt 3 app using express. I have created a sample repository here:

https://github.com/jeremeevans/nuxtnitronode

The goal here is to serve the Nuxt app from a folder, "a-folder", under express, so I would expect to be able to browse to http://localhost:3000/a-folder and see the index page, and go to http://localhost:3000/a-folder/api/health to see the server API result.

The nuxt.config.ts is set to use nitro preset: "node", and the rootDir is set to "a-folder".

The docs for using the nitro node preset say to import handler; however, the generated server/index.mjs does not export handler! Instead I'm doing this (which serves the default Nuxt page always...):

```
import express from 'express'
import { handle } from './.output/server/index.mjs'

const app = express()

app.use('/a-folder', handle)
app.listen(3000)
```

What do I need to do to get it to compile correctly to serve via express? Given the repository I have linked to, I would expect to be able to clone it, then do the following:

1. `npm i`
2. `npm run build`
3. `node .\server.mjs`
4. Browse to http://localhost:3000/a-folder or http://localhost:3000/a-folder/api/health and have them actually work.

**Comment by jeremeevans** [comment]
I updated this to use `router: { base: "a-folder" } }` which I think is closer to as intended. The page loads, and the API endpoint works, but all the assets give 404 errors (when running in dev, or serving from the .\server.mjs)... is there something I need to update so the compiled assets serve correctly??
![image](https://user-images.githubusercontent.com/1747006/147506322-e8eced21-a6f0-46fe-b6a9-45a6f8f1d345.png)

**Reply by jandrad** [reply]
I don't know if this is still useful for you but I added the static resources in express and it is working fine for me.

```
import express from 'express';
import { dirname } from 'path';
import { fileURLToPath } from 'url';
import { handle } from './.output/server/index.mjs';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const app = express();
app.use(
    '/_nuxt',
    express.static(__dirname + '/.output/public/_nuxt', {
        immutable: true,
        maxAge: '1y',
    })
);
app.use(express.static(__dirname + '/.output/public'));
app.use(handle);
app.listen(3000);
```
